### ISSUE TITLE
[Request] Fix Shell Casings in Vice City

### ISSUE THREAD

**Issue by Hans1Poppof** [CookiePLMonster/SilentPatch#46]
In Vice City, when firing the Revolver, it ejects shell casings as if it was a self loading firearm, which it shouldn't happen with this kind of weapon as the casings are contained in the drum, could you make the revolver not eject shell casings as you fire?

**Comment by CookiePLMonster** [comment]
That depends on whether any other weapons are "excluded" from this behaviour too. If they all eject casings it's probably impossible, but if they operate on an exclusion list, then most likely.

I'll mark it as an enhancement for now, I'll be willing to reclassify this as a bug if this behaviour is customizable and just wrongly assigned to the revolver.

**Comment by Hans1Poppof** [comment]
Thank you! I don't know how any of this work internally, maybe if this system exists in the way you suggest it could, it would be used by the flamethrower, sniper rifle, rpg, etc, as they don't eject casings.

**Comment by CookiePLMonster** [comment]
Yeah, this might be done on a per-weapon-type basis... although then again minigun is a heavy weapon, yet it has casings. Perhaps it's done for all "instant hit" weapons.

**Comment by Hans1Poppof** [comment]
Yeah... maybe, checked and the both sniper rifles don't have any shell casings, and they are "instant hit" weapons AFAIK, but their lack of casing effect might have something to do with the fact that they are scoped weapons? don't know to be honest if they could get them as well

**Comment by CookiePLMonster** [comment]
No idea at the moment. I'll keep it open as a request for now but might reclassify it to a bug depending on the findings later. Right now I'm only focusing on stabilization.
